Inference/c++/ernie/inference.cc

```diff
@@ -175,21 +177,40 @@ bool ParseLine(const std::string &line,
   pos_ids.name = "placeholder_1";
   tensors->push_back(pos_ids);
 
-  // sent_ids
-  paddle::PaddleTensor sent_ids;
-  ParseTensor<int64_t>(fields[i++], &sent_ids);
-  sent_ids.name = "placeholder_2";
-  tensors->push_back(sent_ids);
-
-  // input_mask
-  paddle::PaddleTensor input_mask;
-  ParseTensor<float>(fields[i++], &input_mask);
-  input_mask.name = "placeholder_3";
-  tensors->push_back(input_mask);
+  if (inputs_num > 2) {
+    // sent_ids
+    paddle::PaddleTensor sent_ids;
+    ParseTensor<int64_t>(fields[i++], &sent_ids);
+    sent_ids.name = "placeholder_2";
+    tensors->push_back(sent_ids);
+
+    // input_mask
+    paddle::PaddleTensor input_mask;
+    ParseTensor<float>(fields[i++], &input_mask);
+    input_mask.name = "placeholder_3";
+    tensors->push_back(input_mask);
+  }
 
   return true;
 }
 
+static int AssessInputNum(std::ifstream& file) {
+  // Assess number of inptus based on number of colons
+  // present in the first line of the input data.
+  std::string line;
+  std::getline(file, line);
+  file.seekg (0, std::ios::beg);
+
+  int colons_num = std::count(line.begin(), line.end(), ':');
+  int inputs_num = -1;
+  if (colons_num == 2)
```

Comment:
Replace L205-L209 codes with：
int inputs_num =  colons_num== 2 ? colons_num: 4;

Reply:
Done. Thank you for the suggestion.